%%
%% bioRxiv preprint version of the manuscript
%%
\documentclass[11pt]{article}

% Packages for bioRxiv formatting
\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{geometry}
\usepackage{graphicx}
\usepackage{amsmath}
\usepackage{amsfonts}
\usepackage{amssymb}
\usepackage{natbib}
\usepackage{url}
\usepackage{hyperref}
\usepackage{lineno}
\usepackage{xcolor}
\usepackage{fancyhdr}
\usepackage{textcomp} % For additional text symbols
\usepackage{gensymb} % For degree and other symbols

% Page geometry for bioRxiv
\geometry{
    letterpaper,
    margin=1in,
    top=1.2in,
    bottom=1.2in
}

% Line numbers for preprint
\linenumbers
\renewcommand\linenumberfont{\normalfont\tiny\sffamily\color{gray}}

% Header/footer setup
\pagestyle{fancy}
\fancyhf{}
\fancyhead[L]{\small bioRxiv preprint}
\fancyhead[R]{\small \today}
\fancyfoot[C]{\thepage}
\renewcommand{\headrulewidth}{0.4pt}

% Title formatting for bioRxiv
\makeatletter
\renewcommand{\maketitle}{%
    \begin{center}
        {\Large\bfseries \@title \par}
        \vskip 1em
        {\large \@author \par}
        \vskip 0.5em
        {\small \@date \par}
    \end{center}
    \vskip 2em
}
\makeatother

% bioRxiv specific formatting
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt}

% Bibliography style for bioRxiv
\bibliographystyle{unsrt}

% Document metadata
\title{nf-core/viralmetagenome: A Novel Pipeline for Untargeted Viral Genome Reconstruction}

\author{%
    Joon Klaps$^{1,*}$, Philippe Lemey$^{1}$, nf-core community$^{2}$, Liana Eleni Kafetzopoulou$^{1}$ \\[0.5em]
    \small $^1$Rega Institute for Medical Research, Department of Microbiology, Immunology and Transplantation, KU Leuven, Belgium \\[0.5em]
    \small $^2$A full list of contributors can be found at https://nf-co.re/community. \\[0.5em]
    \small $^*$Correspondence: joon.klaps@kuleuven.be
}

\date{\today}

\begin{document}

\maketitle

\vskip 1em

% Abstract section for bioRxiv
\section*{Abstract}
\textbf{Motivation:} Eukaryotic viruses present significant challenges for genome reconstruction and variant analysis due to their extensive diversity and potential genome segmentation. While de novo assembly followed by reference database matching and scaffolding is a commonly used approach, the manual execution of this workflow is extremely time-consuming, particularly due to the extensive reference curation required. Here, we address the critical need for an automated, scalable pipeline that can efficiently handle viral metagenomic analysis without manual intervention.\\
\textbf{Results:} We present nf-core/viralmetagenome, a comprehensive viral metagenomic pipeline for untargeted genome reconstruction and variant analysis of eukaryotic DNA and RNA viruses. Viralmetagenome is implemented as a Nextflow workflow that processes short-read metagenomic samples to automatically detect and assemble viral genomes, while also performing variant analysis. The pipeline features automated reference selection, consensus quality control metrics, comprehensive documentation, and seamless integration with containerization technologies, including Docker and Singularity. We demonstrate the utility and accuracy of our approach through validation on both simulated and real datasets, showing robust performance across diverse viral families in metagenomic samples.\\
\textbf{Availability:} nf-core/viralmetagenome is freely available at https://github.com/nf-core/viralmetagenome with comprehensive documentation at https://nf-co.re/viralmetagenome\\
\textbf{Contact:} \href{joon.klaps@kuleuven.be}{joon.klaps@kuleuven.be}\\
\textbf{Supplementary information:} Supplementary data are available at https://github.com/Joon-Klaps/nf-core-viralmetagenome-manuscript online.\\

\textbf{Keywords:} viralmetagenome, bioinformatic pipeline, nextflow, viral metagenomics, viral assembly, viral variant analysis

\vskip 1em

% Include the shared manuscript content
\input{manuscript-content}

% Bibliography
\bibliography{reference}

\end{document}
